pkg/kube/krt/collection.go

```diff
@@ -0,0 +1,383 @@
+// Copyright Istio Authors
```

Comment:
Future task (maybe even for me): add a sample controller so folks can see how it works

Reply:
https://github.com/istio/istio/pull/45426/files may be what you're considering here.

Reply:
That looks like a full implementation. I'm envisioning something more like https://github.com/istio/istio/blob/master/pkg/kube/controllers/example_test.go

Reply:
The tests provide a lot of cases like this. For a full example it would be nice to have the "write to k8s" part (future PR), though I suppose  that example doesn't.

An almost full example is 

```go
krt.NewCollection(pods, func(ctx krt.HandlerContext, i *corev1.Pod) *SimplePod {
		if i.Status.PodIP == "" {
			return nil
		}
		return &SimplePod{
			Named:   NewNamed(i),
			Labeled: NewLabeled(i.Labels),
			IP:      i.Status.PodIP,
		}
	})
```

which makes it less required to have a example_test.go IMO -- that is needed because its so hard to do the right thing, but a core tenant of `krt` is to make it very easy to do the right thing.

All that being said, I am going to add an example :-)